pandas/core/indexes/period.py

```diff
@@ -818,6 +819,29 @@ def intersection(self, other, sort=False):
         result = self._shallow_copy(np.asarray(i8result, dtype=np.int64), name=res_name)
         return result
 
+    def difference(self, other, sort=None):
```

Comment:
eventtually going to share this?

Reply:
yes, all the PI set operations will end up sharing code